File: backend/app/schemas/education.py
from pydantic import BaseModel, Field


class EducationStage(BaseModel):
    stage_id: str
    name: str
    description: str


class EducationCase(BaseModel):
    case_id: str
    title: str
    summary: str
    analysis: str
    tips: list[str] = Field(default_factory=list)


class EducationStageDetailResponse(BaseModel):
    stage_id: str
    name: str
    description: str
    cases: list[EducationCase] = Field(default_factory=list)


class EducationQuestion(BaseModel):
    question_id: str
    question: str
    options: list[str] = Field(default_factory=list)
    answer: int = 0
    explanation: str = ""
    category: str = ""
    difficulty: str = "beginner"
    fraud_type: str = ""
    source_type: str = "static"


class EducationQuestionPublic(BaseModel):
    question_id: str
    question: str
    options: list[str] = Field(default_factory=list)
    category: str = ""
    difficulty: str = "beginner"
    fraud_type: str = ""
    source_type: str = "static"


class EducationQuestionsResponse(BaseModel):
    items: list[EducationQuestionPublic] = Field(default_factory=list)
    total: int = 0


class SubmitTestRequest(BaseModel):
    question_ids: list[str] = Field(default_factory=list)
    answers: dict[str, int] = Field(default_factory=dict)


class SubmitTestDetail(BaseModel):
    question_id: str
    question: str
    options: list[str] = Field(default_factory=list)
    selected: int | None = None
    correct_answer: int
    is_correct: bool
    explanation: str = ""
    category: str = ""
    difficulty: str = "beginner"
    fraud_type: str = ""
    source_type: str = "static"


class WeaknessItem(BaseModel):
    fraud_type: str
    wrong_count: int = 0
    total: int = 0
    accuracy: float = 0.0
    suggestion: str = ""


class TrendPoint(BaseModel):
    timestamp: str
    score: float
    passed: bool


class SubmitTestResponse(BaseModel):
    total: int = 0
    correct: int = 0
    score: float = 0.0
    passed: bool = False
    details: list[SubmitTestDetail] = Field(default_factory=list)
    risk_profile: str = "medium"
    weaknesses: list[WeaknessItem] = Field(default_factory=list)
    recommended_stage: str = "beginner"
    next_actions: list[str] = Field(default_factory=list)
    summary: str = ""
    recent_trend: list[TrendPoint] = Field(default_factory=list)
    trend_delta: float | None = None
    learning_objective: str = ""
    knowledge_gaps: list[str] = Field(default_factory=list)
    micro_lessons: list[str] = Field(default_factory=list)
    common_mistakes: list[str] = Field(default_factory=list)
    coach_feedback: str = ""
    next_plan: list[str] = Field(default_factory=list)


class EducationCoachRequest(BaseModel):
    question: str = ""
    stage_id: str | None = None
    score: float | None = None
    wrong_topics: list[str] = Field(default_factory=list)


class EducationCoachResponse(BaseModel):
    reply: str = ""
    actions: list[str] = Field(default_factory=list)

File: frontend/src/api/education.ts
import api from './index'

export interface EducationStage {
  stage_id: string
  name: string
  description: string
}

export interface EducationCase {
  case_id: string
  title: string
  summary: string
  analysis: string
  tips: string[]
}

export interface EducationStageDetail {
  stage_id: string
  name: string
  description: string
  cases: EducationCase[]
}

export interface EducationQuestionItem {
  question_id: string
  question: string
  options: string[]
  category: string
  difficulty: string
  fraud_type: string
  source_type: string
}

export interface EducationQuestionsResponse {
  items: EducationQuestionItem[]
  total: number
}

export interface SubmitTestDetail {
  question_id: string
  question: string
  options: string[]
  selected: number | null
  correct_answer: number
  is_correct: boolean
  explanation: string
  category: string
  difficulty: string
  fraud_type: string
  source_type: string
}

export interface WeaknessItem {
  fraud_type: string
  wrong_count: number
  total: number
  accuracy: number
  suggestion: string
}

export interface TrendPoint {
  timestamp: string
  score: number
  passed: boolean
}

export interface SubmitTestResponse {
  total: number
  correct: number
  score: number
  passed: boolean
  details: SubmitTestDetail[]
  risk_profile: 'low' | 'medium' | 'high' | string
  weaknesses: WeaknessItem[]
  recommended_stage: 'beginner' | 'intermediate' | 'advanced' | string
  next_actions: string[]
  summary: string
  recent_trend: TrendPoint[]
  trend_delta: number | null
  learning_objective: string
  knowledge_gaps: string[]
  micro_lessons: string[]
  common_mistakes: string[]
  coach_feedback: string
  next_plan: string[]
}

export interface EducationCoachResponse {
  reply: string
  actions: string[]
}

export const educationApi = {
  async getStages() {
    const { data } = await api.get('/education/stages')
    return data as EducationStage[]
  },

  async getStage(stageId: string) {
    const { data } = await api.get(`/education/stage/${stageId}`)
    return data as EducationStageDetail
  },

  async getQuestions(params?: { count?: number; stageId?: string; refresh?: boolean }) {
    const { data } = await api.get('/education/questions', {
      params: {
        count: params?.count,
        stage_id: params?.stageId,
        refresh: params?.refresh || undefined,
      },
    })
    return data as EducationQuestionsResponse
  },

  async submitTest(payload: { question_ids: string[]; answers: Record<string, number> }) {
    const { data } = await api.post('/education/submit-test', payload)
    return data as SubmitTestResponse
  },

  async askCoach(payload: { question: string; stage_id?: string; score?: number; wrong_topics?: string[] }) {
    const { data } = await api.post('/education/coach', payload)
    return data as EducationCoachResponse
  },
}
